### PR TITLE
Allow passing `None` to `Experiment.axis_query`.

### PR DESCRIPTION
**Pull request by thetorpedodog**
cf. #98

### CODE REVIEW

File: python-spec/src/somacore/experiment.py [hunk 1]
```diff
@@ -47,8 +47,8 @@ def axis_query(
         self: _Self,
         measurement_name: str,
         *,
-        obs_query: query.AxisQuery = query.AxisQuery(),
-        var_query: query.AxisQuery = query.AxisQuery(),
+        obs_query: Optional[query.AxisQuery] = None,
+        var_query: Optional[query.AxisQuery] = None,
```
Comment (atolopko-czi) [inline python-spec/src/somacore/experiment.py:51]: docstring for this method indicates `See :class:`query.ExperimentAxisQuery` for details on usage`.  But in fact I think the problem will still persist in `somacore` code.  If caller of this methods specifies `None`, somacore will not convert that to AxisQuery().  Does the fix need to be pushed down into somacore perhaps? Either way, suggest adding a test in this repo to ensure correct behavior.

**Review by atolopko-czi — CHANGES_REQUESTED** [review]
A concern noted in comments.

**Review by bkmartinjr — APPROVED** [review]
this looks great - ty

**Comment by bkmartinjr** [comment]
> LGTM. Should we also add a corresponding unit test in TileDB-SOMA repo?

_Please_ add a test - the lack of them was why we missed this regression previously.

**Pull request merged**